lib/cc/analyzer/engines_runner.rb

```diff
@@ -13,27 +13,36 @@ def initialize(registry, formatter, source_dir, config, requested_paths = [], co
         @config = config
         @requested_paths = requested_paths
         @container_label = container_label
+        @container_listeners = {}
       end
 
-      def run(container_listener = ContainerListener.new)
-        raise NoEnabledEngines if engines.empty?
+      def each_engine
+        engines.each do |engine|
+          yield(engine)
+        end
+      end
 
+      def run
+        raise NoEnabledEngines if engines.empty?
         @formatter.started
-
-        engines.each { |engine| run_engine(engine, container_listener) }
-
+        each_engine { |engine| run_engine(engine) }
         @formatter.finished
       ensure
         @formatter.close if @formatter.respond_to?(:close)
       end
 
+      def set_container_listener(engine, listener)
+        @container_listeners[engine] = listener
+      end
+
       private
 
       attr_reader :requested_paths
 
-      def run_engine(engine, container_listener)
+      def run_engine(engine)
+        listener = @container_listeners[engine] || ContainerListener.new
```

Comment:
I can see how engines-specific listeners may be valuable (or required) soon, but is it needed to do the stdout stuff you have in this PR? Feels orthogonal.